jbpm-task-assigning/kie-server-client-task-assigning/src/test/java/org/kie/server/client/IntegrationTests.java

```diff
@@ -0,0 +1,247 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.server.client;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.time.StopWatch;
+import org.junit.Before;
+import org.junit.Test;
+import org.kie.server.api.model.instance.ProcessInstance;
+import org.kie.server.api.model.taskassigning.PlanningItem;
+import org.kie.server.api.model.taskassigning.PlanningItemList;
+import org.kie.server.api.model.taskassigning.PlanningTask;
+import org.kie.server.api.model.taskassigning.TaskData;
+import org.kie.server.api.model.taskassigning.TaskInputVariablesReadMode;
+import org.kie.server.client.util.TaskDataReader;
+
+import static org.kie.server.api.model.taskassigning.TaskStatus.InProgress;
+import static org.kie.server.api.model.taskassigning.TaskStatus.Ready;
+import static org.kie.server.api.model.taskassigning.TaskStatus.Reserved;
+import static org.kie.server.api.model.taskassigning.TaskStatus.Suspended;
+
+public class IntegrationTests {
+
+    private static final String SERVER_URL = "http://localhost:8180/kie-server/services/rest/server|http://localhost:8080/kie-server/services/rest/server";
+    //private static final String SERVER_URL = "http://localhost:8180/kie-server/services/rest/server";
+
+    private static String CONTAINER_ID = "com.myspace:test-process:1.0.0-SNAPSHOT";
+    private static String PROCESS_ID = "test-process.Produce_Task_For_HR";
+
+    private static List<Integer> pageSizes = Arrays.asList(3000);
+
+    private Map<String, StopWatch> timeRegistry = new HashMap<>();
+
+    private TaskAssigningRuntimeClient client;
+
+    @Before
+    public void setUp() {
+        timeRegistry.clear();
+        client = newTaskAssigningRuntimeClient();
+    }
+
+    @Test
+    public void findTasksWithTaskAssigningClientAllVariablesReadTime() {
+        pageSizes.forEach(pageSize -> testFindTasksWithTaskAssigningClientTimes(pageSize, TaskInputVariablesReadMode.READ_FOR_ALL));
+    }
+
+    @Test
+    public void findTasksWithTaskAssigningClientVariablesWhenPlanningTaskIsNull() {
+        pageSizes.forEach(pageSize -> testFindTasksWithTaskAssigningClientTimes(pageSize, TaskInputVariablesReadMode.READ_WHEN_PLANNING_TASK_IS_NULL));
+    }
+
+    private void testFindTasksWithTaskAssigningClientTimes(int pageSize, TaskInputVariablesReadMode inputVariablesReadMode) {
+        List<String> status = Arrays.asList(Ready,
```

Comment:
This is a bit confusing. Constants names are usually all upper case.

Reply:
+1 and my apologizes, this test is not part or the product code, just a local test I run for playing a bit. Will be removed as part of the review.